src/token/erc20/interface.cairo

```diff
@@ -18,6 +15,21 @@ trait IERC20<TState> {
     fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
 }
 
+#[starknet::interface]
+trait IERC20Metadata<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn decimals(self: @TState) -> u8;
+}
+
+#[starknet::interface]
+trait IERC20SafeAllowance<TState> {
+    fn increase_allowance(ref self: TState, spender: ContractAddress, added_value: u256) -> bool;
+    fn decrease_allowance(
+        ref self: TState, spender: ContractAddress, subtracted_value: u256
+    ) -> bool;
+}
```

Comment:
In Account we have PublicKeyTrait and PublicKeyCamelTrait. Maybe is worth doing something similar here? The name IERC20SafeAllowance seems to imply that safe allowance comes from EIP20.

Reply:
That's a fair point with the name. I'm not sure about adding `Trait` to the name though, so I changed it to `ISafeAllowance`. Let me know what you think

Reply:
I got you. We need to document a pattern for this as well, cause we have traits ending in Trait (like InternalTrait), and then we have interfaces, and I'm not sure where we should use ISafeAllowance or SafeAllowanceTrait in this case (and the same for PublicKeyTrait vs IPublicKey in account). I think I lean towards using the I for these cases, but then we need to document (agree on) when to use Trait at the end (like in InternalTrait, or potentially other traits).

Naming things :(

Reply:
cc @martriay

Reply:
In the security docs PR I've used ExternalTrait for the external methods that don't belong to an interface, basically because using `generate_trait` is not possible when using `#[embeddable_as(InitializableImpl)]`, I would have used that without manually defining the trait if that were possible. I think that's a good example of when to use Trait at the end (ExternalTrait similar to InternalTrait)

Reply:
i like the interfaces' separation! but it's weird to have some impls with prefix and some without (easy to notice in the docs). bear in mind that although in EIP20 the metadata methods are marked as optional and come from the standard, and the `IERC20Metadata` interface as such is not standard as it is in 721

as of traits vs `ISomething` vs etc, to me traits are cairo/rust language objects only, while interfaces (`ISomething`) are part of a contract's external interface (ABI)

Reply:
Should we have IPublicKey for account instead of PublicKeyTrait then? And also, should we use the ExternalTrait trait for making functions that are not part of an interface as external by using generate_trait?